src/tests/suite/daos_test_common.c

```diff
@@ -1290,12 +1290,13 @@ int verify_state_in_log(char *host, char *log_file, char *state)
 
 		if (fp != NULL)
 			pclose(fp);
-		free(line);
 	}
 
+	D_FREE(line);
```

Comment:
this sounds wrong. you should not use D_FREE on a buffer allocated outside of the library (not with D_ALLOC). line is allocated in getline(); so it should be freed with free.
so this should be replace with:
if (line)
   free(line)

Reply:
just to elaborate. "wrong" is overkill, just not ideal. since D_FREE logs that a free is done, but if one doesn't see a matching D_ALLOC, there will be mismatch. but this is probably fine as Jeff mentioned since this is just in daos_test.